### DISCUSSION TITLE
issue: Default Values are not showing in on Custom Components.

### DISCUSSION THREAD

**Discussion by vikas990** [react-hook-form/react-hook-form#7117] [Q&A]
### Version Number

7.20.2

### Codesandbox/Expo snack

https://codesandbox.io/s/summer-hooks-qtx7r

### Steps to reproduce

please check the code sandbox.

### Expected behaviour

it should show the default values on the custom fields but are not showing.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
when you use `register` api, it's important to forward your input reference with `React.forwardRef` when you build things with your own wrapper component. 

refer to the section: https://react-hook-form.com/get-started#Integratinganexistingform

**Reply by vikas990** [reply]
Hy @bluebill1049, Thank you for the quick response.
I m still struggling with the dropdown part I might be doing it wrong can you help me to clear this part.
https://codesandbox.io/s/summer-hooks-qtx7r?file=/components/Form.js
Thank You.

**Reply by bluebill1049** [reply]
for controlled component use this instead: https://react-hook-form.com/api/usecontroller/controller

**Reply by vikas990** [reply]
Still not able to get why are we using the Controllers for Select but not for InputField.

**Reply by bluebill1049** [reply]
please read the doc, there is a lot of good information there.

> React Hook Form embraces uncontrolled components and native inputs, however it's hard to avoid working with external controlled component such as React-Select, AntD and Material-UI. This wrapper component will make it easier for you to work with them.

https://react-hook-form.com/api/usecontroller/controller

**Reply by bluebill1049** [reply]
register works because ref is forwarded: https://reactjs.org/docs/uncontrolled-components.html#:~:text=In%20a%20controlled%20component%2C%20form,form%20values%20from%20the%20DOM.

controlled: https://reactjs.org/docs/forms.html

**Reply by vikas990** [reply]
hy @bluebill1049 , i think i m doing a very stupid mistake can you please help me to understand what I m doing wrong.
https://codesandbox.io/s/summer-hooks-qtx7r?file=/components/Form.js
